### ISSUE TITLE
Catch unary operator after keyword await as error

### ISSUE THREAD

**Issue by timnew** [babel/eslint-plugin-babel#27]
To invert the boolean value (or any unary operator, such as `~` or `-` for numbers) returned from promise with `await`, we should put the operator before the `await` keyword rather than after. But this is counterintuitive and error-prone.

e.g.

``` js
function returnFalse() {
  return Promise.resolve(false);
}

async function test() {
  if(await !returnFalse()) {
    console.log('ok');
  }  
}

test();
```

We might expect `ok` to be displayed, but it never does. Since we apply `!` on the `promise` return by async function, which is meaningless in most cases. So it would be nice if any code that apply `!` after `await` rather than before.

Here is the correct `test()`:

``` js
async function test() {
  if(! await returnFalse()) {
    console.log('ok');
  }  
}

```

I think it will be a general issue when writing code with `async/await`. And I think linter should check fo r this case.

**Comment by hzoo** [comment]
Closing since this would just be something eslint itself handles (since it's es2017)
